### PR TITLE
Feature/color picker

### PR DESCRIPTION
**Pull request by jheijari**
Spectrum color picker component to divmanazer.
Replacing old color picker used in HeatMap.

### CODE REVIEW

**Comment by data-ux** [comment]
When editing a previously created heatmap layer, current layer color values should be repopulated to the color selection boxes (now gray).

File: packages/framework/bundle/heatmap/bundle.js [hunk 1]
```diff
@@ -41,11 +41,11 @@ Oskari.clazz.define("Oskari.mapframework.bundle.heatmap.HeatmapBundle", function
             "src" : "../../../../bundles/mapping/heatmap/resources/css/style.css"
         }, {
         // NOTE! EXTERNAL LIBRARY!
-            "type": "text/javascript",
-            "src": "../../../../libraries/colpick-jQuery-Color-Picker-master/js/colpick.js"
+            "type" : "text/javascript",
+            "src" : "../../../../libraries/spectrum/spectrum-min.js"
```
Comment (data-ux) [inline packages/framework/bundle/heatmap/bundle.js]: ColorPickerInput.js depends on spectrum, so it should be divmanazer's dependency, not heatmap's

File: packages/mapping/ol3/heatmap/bundle.js [hunk 2]
```diff
@@ -41,11 +41,11 @@ Oskari.clazz.define("Oskari.mapframework.bundle.heatmap.HeatmapBundle", function
             "src" : "../../../../bundles/mapping/heatmap/resources/css/style.css"
         }, {
         // NOTE! EXTERNAL LIBRARY!
-            "type": "text/javascript",
-            "src": "../../../../libraries/colpick-jQuery-Color-Picker-master/js/colpick.js"
+            "type" : "text/javascript",
+            "src" : "../../../../libraries/spectrum/spectrum-min.js"
```
Comment (data-ux) [inline packages/mapping/ol3/heatmap/bundle.js]: Same as above.

File: bundles/mapping/heatmap/HeatmapDialog.js [hunk 3]
```diff
@@ -174,23 +181,25 @@ Oskari.clazz.define('Oskari.mapframework.bundle.heatmap.HeatmapDialog',
...
-                me.colors.push(jQuery(colorBox).attr("color"));
-            });
+                element = jQuery(content).find("input:checked");
+
+            if (element.attr('id') === "customTheme") {
+                this.colors = this.colors.concat(this.getColorSetup());
+            }
+            else {
+                var colorBoxes = jQuery(element[0].parentElement).find('.color-box');
+                colorBoxes.forEach(function (colorBox) {
```
Comment (data-ux) [inline bundles/mapping/heatmap/HeatmapDialog.js]: `colorBoxes` is a jQuery selection, not an Array. It does not have `forEach()` method.

**Comment by data-ux** [comment]
Looks good to me!

File: bundles/framework/divmanazer/component/ColorPickerInput.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+ */
+Oskari.clazz.define('Oskari.userinterface.component.ColorPickerInput',
+
+    /**
+     * @method create called automatically on construction
+     * @param object options colorPicker options for customizing spectrum component
+     */
+    function (options) {
+        var me = this;
+        me._localization = Oskari.getLocalization("DivManazer");
```
Comment (ZakarFin) [inline bundles/framework/divmanazer/component/ColorPickerInput.js]: As the console says: "Oskari.getLocalization() will be removed in future release. Use Oskari.getMsg() instead." Other than that, good for merging.

**Pull request merged**